test/assignment1/username-from-conf-file.h

```diff
@@ -1,34 +1,45 @@
 #include <stdio.h>
 #include <stdlib.h>
+#include <ctype.h>
 
 /**
  * @return the name of the user from the conf file, or
  * an empty string if not found.  Must be freed by the caller
  */
 static inline char *malloc_username_from_conf_file()
 {
-    size_t len = 0;
-    char *buffer = malloc(1);
+    static const uint8_t BASE_LEN = 16;
```

Comment:
@solnus I'm not sure I understand the benefit of using 16 bytes here vs 0.  I like using 0 because it makes it clear that the allocation is happening in getline.  We could use some more comments about how the current logic works though.

Reply:
@dwalkes I had posted a comment on that - but I'm used to gerrit, so maybe I didn't do it quite right. Using malloc(1) causes an allocation that essentially forces getline() to realloc. You might as well start with a reasonable size that has a chance of succeeding without a realloc needed. If you want to force getline to do the alloc, you can pass in null instead of doing malloc(1), but seeing the malloc here is (IMO) slightly clearer that there is dynamic memory which must be freed by someone.

Reply:
> I had posted a comment on that

Yep, just didn't see it before I wrote mine.

> Using malloc(1) causes an allocation that essentially forces getline() to realloc

True, but I think the realloc is going to essentially do nothing because the minimum allocation is probably greater than either.  See [this link](https://prog21.dadgum.com/179.html) for instance.  So maybe if we really cared about performance there would be something to consider here, but I wouldn't make that a priority over readability.  It is subjective whether either is more readable.  I'm personally inclined to look at the 16 and assume I'm seeing a bug that there's not enough space allocated to a long possible username, or make sure that I have test cases which cover usernames longer than 16.  The benefit of the previous implementation is that any username is going to be handled exactly the same way, and we don't need different test cases to cover different username lengths.

>  If you want to force getline to do the alloc, you can pass in null instead of doing malloc(1), but seeing the malloc here is (IMO) slightly clearer that there is dynamic memory which must be freed by someone.

Agreed, I prefer this approach as well.

Reply:
> True, but I think the realloc is going to essentially do nothing because the minimum allocation is probably greater than either. 

Yes, but a resize is still more work than actually required :-)

> I'm personally inclined to look at the 16 and assume I'm seeing a bug

The realloc is non-obvious either way. When I looked at it and saw malloc(1), that initially looked like a bug. If you want to keep the malloc here but always force getline to realloc, then malloc(0) (and a comment) might be a better option.

Reply:
> When I looked at it and saw malloc(1), that initially looked like a bug

malloc(1) is because len is string length, so len=0 corresponds to a 1 byte zero terminated string.  Agree a comment is needed here.

Reply:
@dwalkes I just pushed an update, please check if it is a good compromise

Reply:
Thanks @solnus looks great!
@sarayumanagoli or @akshita-bhasin can you please test this so we can get it merged?  I've opened an issue on https://github.com/cu-ecen-5013/assignment-autotest/issues/11 and added it to our project board.